client/js/src/types.ts

```diff
@@ -238,6 +239,13 @@ export interface LogMessage {
 	log: string;
 	level: "warning" | "info";
 }
+export interface RenderMessage {
+	fn_index: number;
+	data: {
+		components: any[];
+		layout: any;
+	};
```

Comment:
I don't think we can import from js/app into client, because that would be a circular dependency given we import client into js/app? At least that's why I've assumed that we redefined Dependency and Config in both js/app/src/types.ts and client/js/src/types.ts. We also marked Config.components as an `any` type in client/js/src/types.ts instead of using ComponentMeta.
Agree that these definitions should all be merged at some point.